crates/ide_assists/src/handlers/extract_variable.rs

```diff
@@ -146,6 +166,30 @@ fn valid_target_expr(node: SyntaxNode) -> Option<ast::Expr> {
     }
 }
 
+fn get_receiver_type(ctx: &AssistContext, expression: &ast::Expr) -> Option<hir::Type> {
+    let receiver = get_receiver(expression.to_owned())?;
```

Comment:
```suggestion
    let receiver = get_receiver(expression.clone())?;
```
Using `clone` is more idiomatic(at least for our codebase) than using `to_owned` to go from `&T` to `T`